exporter/syslogexporter/exporter.go

```diff
@@ -69,78 +68,30 @@ func newLogsExporter(
 	)
 }
 
-func (se *syslogexporter) logsToMap(record plog.LogRecord) map[string]any {
-	attributes := record.Attributes().AsRaw()
-	return attributes
-}
-
-func (se *syslogexporter) getTimestamp(record plog.LogRecord) time.Time {
-	timestamp := record.Timestamp().AsTime()
-	return timestamp
-}
-
-func (se *syslogexporter) pushLogsData(_ context.Context, ld plog.Logs) error {
-	type droppedResourceRecords struct {
-		resource pcommon.Resource
-		records  []plog.LogRecord
-	}
-	var (
-		errs    []error
-		dropped []droppedResourceRecords
-	)
-	rls := ld.ResourceLogs()
-	for i := 0; i < rls.Len(); i++ {
-		rl := rls.At(i)
-		if droppedRecords, err := se.sendSyslogs(rl); err != nil {
-			dropped = append(dropped, droppedResourceRecords{
-				resource: rl.Resource(),
-				records:  droppedRecords,
-			})
-			errs = append(errs, err)
-		}
-	}
-	if len(dropped) > 0 {
-		ld = plog.NewLogs()
-		for i := range dropped {
-			rls := ld.ResourceLogs().AppendEmpty()
-			logRecords := rls.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
-			dropped[i].resource.MoveTo(rls.Resource())
-			for j := 0; j < len(dropped[i].records); j++ {
-				dropped[i].records[j].MoveTo(logRecords)
+func (se *syslogexporter) pushLogsData(_ context.Context, logs plog.Logs) error {
+	var payload strings.Builder
+	for i := 0; i < logs.ResourceLogs().Len(); i++ {
+		resourceLogs := logs.ResourceLogs().At(i)
+		for j := 0; j < resourceLogs.ScopeLogs().Len(); j++ {
+			scopeLogs := resourceLogs.ScopeLogs().At(j)
+			for k := 0; k < scopeLogs.LogRecords().Len(); k++ {
+				logRecord := scopeLogs.LogRecords().At(k)
+				formatted := se.formatter.format(logRecord)
+				payload.WriteString(formatted)
```

Comment:
should we have a configurable limit on the payload size, now that we're batching log messages?

Reply:
Good point that it should be possible to limit the payload size. Users should be able to use the collector's built-in batching for this. Currently this functionality is provided the [Batch processor](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor). In future, batching hopefully will be a part of the [exporter helper](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/exporterhelper).

Reply:
oh right, I forgot that this is an exporter and not the receiver.